### PR TITLE
Restarting and MPI gtest

### PR DESCRIPTION
**Pull request by weixiong-zheng-berkeley**
Rewrote BART functions with partial testing. Enabled MPI with gtest

### CODE REVIEW

File: src/aqdata/tests/lsgc_test.cc [hunk 1]
```diff
@@ -40,7 +40,7 @@ void LSGCTest::AQDataTest() {
   lsgc_ptr->MakeAQ();
   auto wi = lsgc_ptr->GetAQWeights();
   auto omega_i = lsgc_ptr->GetAQDirs();
-  for (unsigned int i=0; i<wi.size(); ++i)
+  for (int i=0; i<wi.size(); ++i)
```
Comment (jsrehak) [inline src/aqdata/tests/lsgc_test.cc]: It occurs to me, after our discussion of `int` vs `unsigned`, we should consider using `std::size_t i = 0` for these loops.

Comment (jsrehak) [reply src/aqdata/tests/lsgc_test.cc]: Ah interesting! I should have checked the style guide, that works, sorry!

For vectors, at least, `std::vector::size_type` _is_ just a typedef of `size_t`:
```
class vector :
...
      typedef size_t					size_type;
...
```

Comment (weixiong-zheng-berkeley) [reply src/aqdata/tests/lsgc_test.cc]: Thanks for the good question!

I will look at it later. If that's true, I don't understand about the compiler warning when trying `size_t`. But anyways, if we stick to Google Style Guide, we'll use `int` for sure.

**Comment by jsrehak** [comment]
Looks good, I mostly reviewed it for style, because I don't know exactly how everything works yet.

**Pull request merged**